src/main/scala/esl/FSConnection.scala

```diff
@@ -837,6 +837,36 @@ abstract class FSConnection extends StrictLogging {
     * @return EventMessage
     */
   private def handleFSEventMessage(eventMessage: EventMessage): EventMessage = {
+    def completeAndRemoveFromMap(command: FSExecuteApp, executeComplete: Promise[EventMessage]): Unit = {
```

Comment:
def completeAndRemoveFromMap(command: FSExecuteApp, executeComplete: Promise[EventMessage], canRemove:Boolean): Unit = {
      executeComplete.complete(Success(eventMessage))
      if (canRemove) {
        eventMap.remove(command.eventUuid)
        adapter.info(
          logMarker,
          s"""handleFSEventMessage for app id ${command.eventUuid} Removing entry
             |>> TYPE
             |EVENT ${eventMessage.eventName.getOrElse("NA")}
             |>> HEADERS
             |${
            eventMessage.headers
              .map(h => h._1 + " : " + h._2)
              .mkString(space, "\n" + space, "")
          }
             |>> BODY
             |${eventMessage.body}
             |>> MAP is below
             |${
            eventMap.map({ case (key, value) =>
                s"""appId: $key
                   |command
                   |$value""".stripMargin
              })
              .mkString("\n")
          }""".stripMargin
        )
      }
    }